src/Castle.Core/DynamicProxy/Generators/MethodSignatureComparer.cs

```diff
@@ -128,5 +132,36 @@ private bool EqualNames(MethodInfo x, MethodInfo y)
 		{
 			return x.Name == y.Name;
 		}
+
+		private bool GenericArugmentsEqual(Type x, Type y)
+		{
+			var xArgs = x.GetGenericArguments();
+			var yArgs = y.GetGenericArguments();
+
+			if (xArgs.Length != yArgs.Length)
+			{
+				return false;
+			}
+
+			for (var i = 0; i < xArgs.Length; ++i)
+			{
+				if (xArgs[i].GetTypeInfo().IsGenericParameter != yArgs[i].GetTypeInfo().IsGenericParameter)
```

Comment:
Could this loop just call `EqualSignatureTypes(xArgs[i], yArgs[i])`? This probably means this method isn't needed and the recursion could happen in `EqualSignatureTypes` instead.

If this method stays it is spelt wrong.